OKTA-459955 Embedded SPA + Auth JS for Vue.js (#3010)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-authjs/main/vue/sign-in-form.md

### Set up the Okta configuration settings

Before you code your forms and routes, use the required [configuration settings](#okta-org-app-integration-configuration-settings) to initialize your Okta Auth JS instance:

* `clientId`: Your client ID &mdash; `${yourClientId}`

> * `issuer`: The authorization server in your Okta org &mdash; `${yourIssuer}`

Author: I was trying to correspond this value to the variable in the next code snippet. But I think you're right. I'll add the full URI in both places for clarity. Thanks!

> This section guides you to build a basic password-only sign-in use case for your app. This use case is outlined in the following sequence diagram with your single-page app as the client:

Reviewer: ```suggestion
This section helps you build a basic password-only sign-in use case for your app. This use case is outlined in the following sequence diagram with your single-page app as the client:
```

<!-- Source for image. Generated using http://www.plantuml.com/plantuml/uml/

@startuml
skinparam monochrome true
actor "Resource Owner (User)" as user
participant "Client (SPA)" as client
participant "Auth JS (SDK)" as sdk
participant "Authorization server (Okta)" as okta

autonumber "<b>#."
user -> client: Navigate to app sign-in page
client -> client: Display sign-in page, instantiate OktaAuth()
user -> client: Enter credentials
client -> sdk: Call idx.authenticate(username,password)
sdk -> okta: API request to authenticate user
okta -> sdk: Return Auth response
sdk -> client: Return tokens and idxStatus.SUCCESS
client -> client: For idxStatus.SUCCESS, store tokens in browser storage
client -> user: Direct user to authenticated page
client -> client: For idxStatus.SUCCESS, store tokens in browser storage
client -> sdk: (Optional) Call token.getUserInfo() to get user info
sdk -> okta: (Optional) API request to get user info
okta -> sdk: (Optional) Return user info response
sdk -> client: (Optional) Return user info
client -> user:  (Optional) Display required user info
@enduml

-->

> The steps to add Okta authentication described in the following sections focus on the interaction between your client app, the user, and the Auth JS SDK.

Reviewer: is "add Okta authentication" necessary to complete the meaning?

```suggestion
The steps in the following sections focus on the interaction between your client app, the user, and the Auth JS SDK.
```

### Set up the Okta configuration settings

Before you code your forms and routes, use the required [configuration settings](#okta-org-app-integration-configuration-settings) to initialize your Okta Auth JS instance:

* `clientId`: Your client ID &mdash; `${yourClientId}`
* `issuer`: The authorization server in your Okta org (for example, `https://${yourOktaDomain}/oauth2/default`)
* `scopes`: The required OAuth 2.0 [scopes](/docs/reference/api/oidc/#scopes) for your app
* `redirectUri`: Set your callback redirect URI. This value must be configured in your Okta app **Sign-in redirect URIs** and **Trusted Origins** lists.

You can create a `src/config.js` file to define your configuration settings. For example:

```js
export default {
  oidc: {
    clientId: '${yourClientId}',
    issuer: 'https://${yourOktaDomain}/oauth2/default',
    redirectUri: '${yourLocalAppDomain}/login/callback',
    scopes: ['openid', 'profile', 'email'],
    tokenManager: {
      storage: 'localStorage'
    }
  }
}
```

> > **Note**: See [Okta Auth JS configuration reference](https://github.com/okta/okta-auth-js#configuration-reference) for additional Auth JS client configurations.

Reviewer: ```suggestion
> **Note**: See the [Okta Auth JS configuration reference](https://github.com/okta/okta-auth-js#configuration-reference) for additional Auth JS client configurations.
```
